api/types.go

```diff
@@ -157,6 +157,32 @@ func (tr Timerange) IsValid() bool {
 		tr.Start <= tr.End
 }
 
+func rounder(x, r int64) int64 {
+	if x < 0 {
+		return -rounder(-x, r)
+	}
+	// This performs a round
+	return (x + r/2) / r * r
```

Comment:
Since `x / r * r = (x / r) * r`, and `x / r` truncates towards zero, this accomplishes a round (towards 0). A round "towards nearest" (with halfways rounding up) is accomplished by the `+ r/2`.

Unfortunately, the golang `math` (or any other I know of) library does not provide an equivalent...

It's definitely not the most clear expression...